doc/user/content/transform-data/patterns/partition-by.md

```diff
@@ -0,0 +1,187 @@
+---
+title: "Partitioning and filter pushdown"
+description: "Declare how collections are stored."
+aliases:
+  - /guides/partition-by/
+  - /sql/patterns/partition-by/
+menu:
+  main:
+    parent: 'sql-patterns'
+---
+
+[//]: # "TODO link to the source table docs once that feature is documented."
+
+{{< private-preview />}}
+
+A few types of Materialize collections are durably written to storage: [materialized views](/sql/create-materialized-view/), [tables](/sql/create-table), and [sources](/sql/create-source).
+
+Internally, Materialize stores these durable collections in an [LSM-tree](https://en.wikipedia.org/wiki/Log-structured_merge-tree)-like structure. Each collection is made up of a set of
```

Comment:
This is technically correct, but I wonder if we should simplify the explanation? If we just skipped the first 1 and 1/2 sencence and said something like
> Data is sorted and then partitioned up into individual **parts**, ...

And then later

>Using the `PARTITION BY` option, you can specify the internal ordering that Materialize will use to sort, partition, and store the data.

Reply:
The thing I worry about is questions like: "I know my query only matches a single record; if all the data is sorted, why does my query need to fetch more than one part?" Really hard to explain without the concept of "runs" here.

I'm definitely sensitive to this point - so far these are implementation details we haven't spoken much about publicly, and it makes sense to keep the description as light as possible. But I don't want to give the impression that everything is one sorted set of parts...

Reply:
I see. Would it make sense to add a tiny example at the end that helps clarify the concepts and shows how filter pushdown can skip certain runs?